Permit lines to be hidden from consideration for wrong-import-order and wrong-import-position.

erikwright opened the pull request:
@honkfestival PTAL. I'll point our requirements at this fork, and hopefully get it merged upstream.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/shopify/pylint/1)
<!-- Reviewable:end -->

erikwright committed:
Optimize effect of disabling import warnings

Disabling 'wrong-import-order' or 'wrong-import-position' for a single line now prevents that line from triggering violations on subsequent lines.

honkfestival commented on pylint/test/functional/disable_wrong_import_order.py:
```diff
@@ -0,0 +1,11 @@
+import os.path
+import sys
+from astroid import are_exclusive
+import first_party  # [ungrouped-imports]
```
Does the comment participate in their testing infrastructure?

erikwright replied:
Yeah it seems to tell the tester that the specified error is expected on the commented line. The actual message output appears in the associated `.txt` file. It seems maybe a bit redundant at first glance.

honkfestival commented on pylint/checkers/imports.py:
```diff
@@ -527,8 +531,8 @@ def _check_imports_order(self, node):
-        extern_not_nested = []
-        local_not_nested = []
+        extern_not_ignored = []
+        local_not_ignored = []
```
👍

honkfestival reviewed (approved):
LGTM

erikwright committed:
Optimize effect of disabling ungrouped-imports.

Disabling 'ungrouped-imports' for a single line now prevents that line from triggering violations on subsequent lines.

JasonMWhite commented:
There is already a branch 1.6.5 on the source project. Could we branch off that instead? There are other improvements from that branch we could use.

erikwright commented:
I've also pushed to `shopify-1.6`, which is off the tip of their 1.6.5
branch I believe. Yes we can use that.

On Fri, Feb 24, 2017 at 4:59 PM, Jason White <notifications@github.com>
wrote:

> There is already a branch 1.6.5 on the source project. Could we branch off
> that instead? There are other improvements from that branch we could use.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Shopify/pylint/pull/1#issuecomment-282417009>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AALat7qZsmF9CqXMGS6stL0iCdo20an3ks5rf1K9gaJpZM4MHuYn>
> .
>

JasonMWhite commented:
`1.6-shopify` is significantly behind master (~500 commits) and doesn't include the improvement I was looking at (name checking when creating new types from `typing`)
